File: frontend/BusinessFlow/src/app/modules/support/services/sla-policy.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService, PagedResponse } from '../../../core/services/api.service';
import { SLAPolicy, SLAPolicyRequest } from '../models/support.model';

@Injectable({ providedIn: 'root' })
export class SLAPolicyService {
  private readonly endpoint = '/support/sla-policies';
  constructor(private api: ApiService) {}

  list(page = 0, size = 20): Observable<PagedResponse<SLAPolicy>> { return this.api.getPaged<SLAPolicy>(this.endpoint, page, size); }
  active(): Observable<SLAPolicy[]> { return this.api.get<SLAPolicy[]>(`${this.endpoint}/active`); }
  getById(id: number): Observable<SLAPolicy> { return this.api.get<SLAPolicy>(`${this.endpoint}/${id}`); }
  getByPriority(priority: string): Observable<SLAPolicy> { return this.api.get<SLAPolicy>(`${this.endpoint}/priority/${priority}`); }
  create(payload: SLAPolicyRequest): Observable<SLAPolicy> { return this.api.post<SLAPolicy>(this.endpoint, payload); }
  update(id: number, payload: SLAPolicyRequest): Observable<SLAPolicy> { return this.api.patch<SLAPolicy>(`${this.endpoint}/${id}`, payload); }
  // Backend expects @RequestParam boolean active, and returns no body
  updateStatus(id: number, active: boolean): Observable<void> { return this.api.patch<void>(`${this.endpoint}/${id}/status?active=${active}`, {}); }
  delete(id: number): Observable<SLAPolicy> { return this.api.delete<SLAPolicy>(`${this.endpoint}/${id}`); }
}

File: backend/ZuhooCMS/src/main/java/com/zuhoocms/modules/support/sla/SLAPolicyService.java
package com.zuhoocms.modules.support.sla;

import com.zuhoocms.modules.support.ticket.TicketPriority;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.List;

public interface SLAPolicyService {
    SLAPolicyResponse create(SLAPolicyRequest request);
    SLAPolicyResponse getById(Long id);
    SLAPolicyResponse getByPriority(TicketPriority priority);
    Page<SLAPolicyResponse> getAll(Pageable pageable);
    List<SLAPolicyResponse> getActive();
    SLAPolicyResponse update(Long id, SLAPolicyRequest request);
    void updateStatus(Long id, boolean active);
    SLAPolicyResponse delete(Long id);
}
